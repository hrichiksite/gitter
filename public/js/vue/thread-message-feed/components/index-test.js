const mount = require('../../__test__/vuex-mount');
const {
  createSerializedMessageFixture,
  createSerializedRoomFixture
} = require('../../__test__/fixture-helpers');
const { default: Index } = require('./index.vue');

describe('thread-message-feed', () => {
  const addDisplayedRoom = (state, roomMember = true) => {
    const displayedRoom = createSerializedRoomFixture({ roomMember });
    state.roomMap = { [displayedRoom.id]: displayedRoom };
    state.displayedRoomId = displayedRoom.id;
  };
  const addDefaultUser = state => {
    state.user = { displayName: 'John Smith' };
  };
  const addParentMessage = state => {
    const parentMessage = createSerializedMessageFixture();
    state.messageMap = { [parentMessage.id]: parentMessage };
    state.threadMessageFeed.parentId = parentMessage.id;
  };
  const setupDefaultState = state => {
    addDisplayedRoom(state);
    addDefaultUser(state);
    addParentMessage(state);
  };

  it('closed - matches snapshot', async () => {
    const { wrapper } = mount(Index, {}, store => {
      store.state.threadMessageFeed.isVisible = false;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('not room member - matches snapshot', async () => {
    const { wrapper } = mount(Index, {}, store => {
      addParentMessage(store.state);
      addDisplayedRoom(store.state, false);
      store.state.threadMessageFeed.isVisible = true;
      const parentId = store.state.threadMessageFeed.parentId;
      store.state.messageMap = {
        ...store.state.messageMap,
        1: createSerializedMessageFixture({ id: '1', parentId }),
        2: createSerializedMessageFixture({ id: '2', parentId })
      };
      store.state.roomMap[store.state.displayedRoomId].roomMember = false;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('child messages', () => {
    it('opened - matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        setupDefaultState(store.state);
        store.state.threadMessageFeed.isVisible = true;
        const parentId = store.state.threadMessageFeed.parentId;
        store.state.messageMap = {
          ...store.state.messageMap,
          1: createSerializedMessageFixture({ id: '1', parentId }),
          2: createSerializedMessageFixture({ id: '2', parentId })
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('loading - matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        setupDefaultState(store.state);
        store.state.threadMessageFeed.isVisible = true;
        store.state.threadMessageFeed.childMessagesRequest.loading = true;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('error - matches snapshot', () => {
      const { wrapper } = mount(Index, {}, store => {
        setupDefaultState(store.state);
        store.state.threadMessageFeed.isVisible = true;
        store.state.threadMessageFeed.childMessagesRequest.error = true;
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('dark theme - matches snapshot', () => {
    const { wrapper } = mount(Index, {}, store => {
      setupDefaultState(store.state);
      store.state.threadMessageFeed.isVisible = true;
      store.state.darkTheme = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('missing parent message - matches snapshot', () => {
    const { wrapper } = mount(Index, {}, store => {
      setupDefaultState(store.state);
      store.state.threadMessageFeed.isVisible = true;
      store.state.messageMap = {};
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
