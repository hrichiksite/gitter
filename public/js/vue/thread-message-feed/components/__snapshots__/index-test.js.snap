// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed child messages error - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="content"
      role="log"
    >
      <div
        class="chat-messages"
        role="list"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <div
          class="error-text error-box"
        >
          
          Error: Thread messages can't be fetched.
        
        </div>
         
        <intersect-stub>
          <div />
        </intersect-stub>
          
        <intersect-stub>
          <div />
        </intersect-stub>
      </div>
       
      <chat-input-stub
        roommember="true"
        thread="true"
        user="[object Object]"
      />
    </section>
  </section>
</div>
`;

exports[`thread-message-feed child messages loading - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="content"
      role="log"
    >
      <div
        class="chat-messages"
        role="list"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <div
          class="loading-message"
        >
          
          Loading thread 
          <loading-spinner-stub />
        </div>
         
        <intersect-stub>
          <div />
        </intersect-stub>
          
        <intersect-stub>
          <div />
        </intersect-stub>
      </div>
       
      <chat-input-stub
        roommember="true"
        thread="true"
        user="[object Object]"
      />
    </section>
  </section>
</div>
`;

exports[`thread-message-feed child messages opened - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="content"
      role="log"
    >
      <div
        class="chat-messages"
        role="list"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <!---->
         
        <intersect-stub>
          <div />
        </intersect-stub>
         
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <intersect-stub>
          <div />
        </intersect-stub>
      </div>
       
      <chat-input-stub
        roommember="true"
        thread="true"
        user="[object Object]"
      />
    </section>
  </section>
</div>
`;

exports[`thread-message-feed closed - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root"
>
  <!---->
</div>
`;

exports[`thread-message-feed dark theme - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened dark-theme"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="content"
      role="log"
    >
      <div
        class="chat-messages"
        role="list"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <!---->
         
        <intersect-stub>
          <div />
        </intersect-stub>
          
        <intersect-stub>
          <div />
        </intersect-stub>
      </div>
       
      <chat-input-stub
        roommember="true"
        thread="true"
        user="[object Object]"
      />
    </section>
  </section>
</div>
`;

exports[`thread-message-feed missing parent message - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      class="content"
    >
      <span
        class="error-text error-box"
      >
        
        Error: The message for this thread is unavailable. It was probably deleted.
      
      </span>
    </section>
  </section>
</div>
`;

exports[`thread-message-feed not room member - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="content"
      role="log"
    >
      <div
        class="chat-messages"
        role="list"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <!---->
         
        <intersect-stub>
          <div />
        </intersect-stub>
         
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
         
        <intersect-stub>
          <div />
        </intersect-stub>
      </div>
       
      <join-or-sign-in-button-stub />
    </section>
  </section>
</div>
`;
